src/server/game/Spells/SpellEffects.cpp

```diff
@@ -5919,3 +5919,16 @@ void Spell::EffectLearnTransmogIllusion()
 
     player->GetSession()->GetCollectionMgr()->AddTransmogIllusion(illusionId);
 }
+
+void Spell::EffectSetChromieTime()
+{
+    if (effectHandleMode != SPELL_EFFECT_HANDLE_HIT_TARGET)
+        return;
+
+    Player* caster = m_caster->ToPlayer();
+    if (!caster)
+        return;
+
+    UiChromieTimeExpansionInfoEntry const* expansion = sUiChromieTimeExpansionInfoStore.LookupEntry(effectInfo->MiscValue);
```

Comment:
```suggestion
    UiChromieTimeExpansionInfoEntry const* expansion = sUiChromieTimeExpansionInfoStore.LookupEntry(effectInfo->MiscValue);
    if (!expansion)
        return;
```

Reply:
thanks for reminding me to handle select the present xd